torch/onnx/_internal/exporter.py

```diff
@@ -186,6 +222,30 @@ def resolve(value: Optional[T], fallback: Union[T, Callable[[], T]]) -> T:
                     assert hasattr(self, key), f"Unresolved option '{key}'"
 
 
+@contextlib.contextmanager
+def enable_fake_mode():
+    """Enable fake mode for the duration of the context.
+
+    User input and model are converted to fake tensors, which means they
+    do not carry actual data, but only shape information.
+
+    It must be used when exporting models that are too large to fit into memory.
+    """
+    from torch._subclasses import fake_tensor
+    from torch.fx.experimental.symbolic_shapes import ShapeEnv
+
+    fake_mode = fake_tensor.FakeTensorMode(
+        allow_non_fake_inputs=False,
+        shape_env=ShapeEnv(
+            allow_scalar_outputs=False, allow_dynamic_output_shape_ops=False
+        ),
+    )
+    fake_context = ONNXFakeContext(fake_mode=fake_mode)
+    torch_patcher = patcher.ONNXTorchPatcher()
```

Comment:
Is this necessary to use `torch_patcher` here? After reading code at `ONNXTorchPatcher`, the two relevant patches for us is `torch.load` and `torch._util._rebuild_tensor`.

- First patch stores `f` into `torch_patcher.paths` when calling torch.load. However this is unused here. Is there plan to utilize the info here such that no additional argument regarding state_dict is needed when doing final serialization?
- Second patch hijack tensor creation and create fake tensor instead. Is this not covered by `fake_mode` alone?

Reply:
Good catch. This used to be needed here when `enable_fake_model(model_state_dict: Any)` had a `model_state_dict` argument to pre-process serialization.

now `ExportOutput.save` receives a `model_state_dict` and call `ONNXTorchPatcher`